cclib/parser/turbomoleparser.py

```diff
@@ -64,9 +63,48 @@ def __init__(self, *args, **kwargs):
         self.hours_regex = re.compile(r"([0-9.]*) hours")
         self.minutes_regex = re.compile(r"([0-9.]*) minutes")
         self.seconds_regex = re.compile(r"([0-9.]*) seconds")
-
-        # A list of previous lines to allow look-behind functionality.
-        self.last_lines = collections.deque([""] * 10, 10)
+        
+    def sort_input(self, file_names: list) -> list:

```

Comment:
Putting this alongside the parser is ok, but the reason it was a bare function is that `self` isn't necessary.  I could see it being an abstract class method in the future, for other parsers that can take multiple files like Molpro, but a top-level (private) function would work just as well.

Same comment about using `list` as a type annotation.  (It translates to `typing.List[Any]` which isn't great.)

Reply:
Yeah ok I see your point. My main motivation for making it a class method was to take advantage of inheritance to automatically pick the correct sorting function, seeing as how we've already got the correct parser class at this point. Should make future additions easier because there's no need to maintain a table of functions or similar.

I'll change it to a classmethod, although if we'd rather go with non-class function I'm not totally opposed either.

Reply:
Also, I actually rely on `cclib.io.ccio.sort_turbomole_outputs()` in my own code, which this change obviously breaks. For me that's not a problem, but for others who might rely on it this is a non backwards compatible change. Think it's worth adding back `cclib.io.ccio.sort_turbomole_outputs()` as an alias to the new class method (or whatever we end up going with)?

Reply:
Breaking a function like that across a major or even minor (but not patch) release is ok, at least when you consider how many people were likely using it compared to `ccread` or `ccopen`.

Keeping it as an alias is ok, and if we had an official way of doing deprecation (it's never come up) we could get rid of it after 1.8.x.

But I am thinking the method is the better approach.  It just isn't clear to me yet which one of instance/class/static it should be.  Class is just a good compromise.

Reply:
Cool, I'll add an alias and make it emit a warning on first use, that way at least the user/developer is aware they're using a deprecated function.

Class or abstract makes the most sens I think, and to me class always feels more natural than abstract...